### PR TITLE
build: add Taskfile.yml to use task as a build tool

### PR DESCRIPTION
**Pull request by eitsupi**
A part of #556

Allows use of `task` instead of `make`.
Most of the content was copied from the prqlr repository.

### CODE REVIEW

**Comment by eitsupi** [comment]
At the moment, only the main targets have been ported from the Makefile, and the minor targets have not yet been ported.
But building and testing the entire package should work. Like `task build-all`, `task build-website`, etc.

@etiennebacher Could you try this?
Task should work on Windows since it doesn't depend on Unix shell, but I haven't tested it here.

**Comment by etiennebacher** [comment]
Thanks @eitsupi, I confirm it works fine on Windows once I downloaded `Task` (I just tried `build-all` and `setup-dev`). I'll need more time to see if it fixes #613 but at least the basic usage works

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    cmds:
+      - R CMD INSTALL --no-multiarch --with-keep.source .
+
+  build-website:
+    desc: Build the website.
+    sources:
+      - man/*.Rd
+      - altdoc/altdoc_preprocessing.R
+      - altdoc/*.yml
+      - vignettes/*.Rmd
```
Comment (eitsupi) [inline Taskfile.yml:229]: We can use the `{{.VIGNETTES}}` variable instead of this.

**Comment by eitsupi** [comment]
I believe I have ported the full functionality of the Makefile.

```sh
task setup-dev
task build-all
task build-website
```

**Review by etiennebacher — CHANGES_REQUESTED** [review]
I think the documentation workflow needs to be updated as well to use task instead

**Comment by eitsupi** [comment]
> I think the documentation workflow needs to be updated as well to use task instead

I'd like to do that in a follow-up PR. This PR includes an update to Rbuildignore, so it triggers the R workflow needlessly.

**Review by etiennebacher — APPROVED** [review]
Thanks @eitsupi

**Comment by eitsupi** [comment]
Thanks for taking a look at this.

**Comment by etiennebacher** [comment]
@eitsupi I use the taskfile more regularly now and the Rust crates are only compiled once with `task build-all`, thanks!

**Pull request merged**